client.go

```diff
@@ -929,13 +929,12 @@ func (client *client) tryRefreshMetadata(topics []string, attemptsRemaining int,
 
 // if no fatal error, returns a list of topics that need retrying due to ErrLeaderNotAvailable
 func (client *client) updateMetadata(data *MetadataResponse, allKnownMetaData bool) (retry bool, err error) {
-	if client.Closed() {
-		return
-	}
-
 	client.lock.Lock()
 	defer client.lock.Unlock()
 
+	if client.brokers == nil {
+		return
+	}
```

Comment:
I don't understand how the proposed change would prevent a panic? Can you provide backtrace output from when you've hit a problem here? Alternatively can you demonstrate via a unittest what state causes a panic and how the change prevents that?